File: kotlin-back/src/main/kotlin/com/example/kotlinback/member/repository/MemberRepository.java
package com.example.kotlinback.member.repository;

import com.example.kotlinback.member.entity.Member;

import java.util.Optional;

public interface MemberRepository {
    Member save(Member member);

    Optional<Member> findByUsername(String username);

    Optional<Member> findByNickname(String nickname);

    Optional<Member> findById(String id);

    Optional<Member> findByEmail(String email);
}

File: kotlin-back/src/main/kotlin/com/example/kotlinback/member/repository/MemberRepository.kt
package com.example.kotlinback.member.repository

import com.example.kotlinback.k.member.entity.Member
import java.util.*

interface MemberRepository {
    fun save(member: Member?): Member?
    fun findByUsername(username: String?): Optional<Member?>?
    fun findByNickname(nickname: String?): Optional<Member?>?
    fun findById(id: String?): Optional<Member?>?
    fun findByEmail(email: String?): Optional<Member?>
}
